apps/studio/pages/project/[ref]/functions/new.tsx

```diff
@@ -0,0 +1,286 @@
+  const [savedCode, setSavedCode] = useState<string>('')
+
+  const { mutateAsync: deployFunction, isLoading: isDeploying } = useEdgeFunctionDeployMutation({
+    onSuccess: () => {
+      toast.success('Successfully deployed edge function')
+      if (ref && functionName) {
+        router.push(`/project/${ref}/functions/${functionName}/details`)
+      }
+    },
+  })
+
+  const onDeploy = async () => {
+    if (isDeploying || !ref || !functionName) return
+
+    try {
+      await deployFunction({
+        projectRef: ref,
+        metadata: {
+          entrypoint_path: 'index.ts',
+          name: functionName,
+          verify_jwt: true,
+        },
+        files: files.map(({ name, content }) => ({ name, content })),
+      })
+    } catch (error) {
+      toast.error(
+        `Failed to deploy function: ${error instanceof Error ? error.message : 'Unknown error'}`
+      )
+    }
+  }
+
+  const handleChat = () => {
+    const selectedFile = files.find((f) => f.selected) ?? files[0]
+    setAiAssistantPanel({
+      open: true,
+      sqlSnippets: [selectedFile.content],
+      initialInput: 'Help me understand and improve this edge function...',
+      suggestions: {
+        title:
+          'I can help you understand and improve your edge function. Here are a few example prompts to get you started:',
+        prompts: [
+          'Explain what this function does...',
+          'Help me optimize this function...',
+          'Show me how to add more features...',
+          'Help me handle errors better...',
+        ],
+      },
+    })
+  }
+
+  const onSelectTemplate = (templateValue: string) => {
+    const template = EDGE_FUNCTION_TEMPLATES.find((t) => t.value === templateValue)
+    if (template) {
+      setFiles((prev) =>
+        prev.map((file) => (file.selected ? { ...file, content: template.content } : file))
+      )
+      setOpen(false)
+    }
+  }
+
+  const handleTemplateMouseEnter = (content: string) => {
+    if (!isPreviewingTemplate) {
+      const selectedFile = files.find((f) => f.selected) ?? files[0]
+      setSavedCode(selectedFile.content)
+    }
+    setIsPreviewingTemplate(true)
+    setFiles((prev) => prev.map((file) => (file.selected ? { ...file, content } : file)))
+  }
+
+  const handleTemplateMouseLeave = () => {
+    if (isPreviewingTemplate) {
+      setIsPreviewingTemplate(false)
+      setFiles((prev) =>
+        prev.map((file) => (file.selected ? { ...file, content: savedCode } : file))
+      )
+    }
+  }
+
+  return (
+    <PageLayout
+      size="full"
+      isCompact
+      breadcrumbs={[
+        {
+          label: 'Edge Functions',
+          href: `/project/${ref}/functions`,
+        },
+        {
+          element: (
+            <div className="flex items-center gap-2">
+              <Label_Shadcn_ htmlFor="function-name" className="text-foreground-light sr-only">
+                Function name
+              </Label_Shadcn_>
+              <Input_Shadcn_
```

Comment:
feedback: this wasn't very clear imo to name the function from here
I was confused for a bit when the deploy function button was disabled, but I couldn't figure out why
![image](https://github.com/user-attachments/assets/ec50229a-0159-437f-bef6-57f68b0640a5)